common/kvstore/src/test/java/org/apache/spark/util/kvstore/ArrayKeyIndexType.java

```diff
@@ -38,7 +38,7 @@ public boolean equals(Object o) {
 
   @Override
   public int hashCode() {
-    return key.hashCode();
+    return Arrays.hashCode(key);
```

Comment:
Given that equals() uses id too, how about `Arrays.hashCode(key) ^ Arrays.hashCode(id)`?

Reply:
That might be the reason of participating only key, but your suggestion would be symmetrical and no big deal (as it had been completely broken). Thanks for the suggestion! Applied.